client.go

```diff
@@ -1,39 +1,116 @@
 package kinesis
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
 	"os"
+	"strings"
+	"time"
 )
 
 const (
 	ACCESS_ENV_KEY  = "AWS_ACCESS_KEY"
 	SECRET_ENV_KEY  = "AWS_SECRET_KEY"
 	REGION_ENV_NAME = "AWS_REGION_NAME"
+
```

Comment:
Consider updating the README with instructions on how to leverage this new functionality.